From 65b467c621227c86a630916f3741b20f7c6954df Mon Sep 17 00:00:00 2001
From: Shashank Sharma <shashank.sharma@intel.com>
Date: Fri, 8 Mar 2019 20:54:19 +0530
Subject: [PATCH 19/27] pixel-formats: Add P010 in pixel formats

This patch adds details for P010 pixel format, in the formats list.
This will allow us to play 10-bit HDR P010 videos.

Signed-off-by: Shashank Sharma <shashank.sharma@intel.com>
Signed-off-by: alex.chae <alex.chae@intel.com>
---
 libweston/pixel-formats.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/libweston/pixel-formats.c b/libweston/pixel-formats.c
index 79dc709c..5d8da27f 100644
--- a/libweston/pixel-formats.c
+++ b/libweston/pixel-formats.c
@@ -52,6 +52,10 @@
 #define SAMPLER_TYPE(type) .sampler_type = 0
 #endif
 
+#ifndef DRM_FORMAT_P010
+#define DRM_FORMAT_P010         fourcc_code('P', '0', '1', '0') /* 2x2 subsampled Cb:Cr plane 10 bits per channel */
+#endif
+
 #define DRM_FORMAT(f) .format = DRM_FORMAT_ ## f, .drm_format_name = #f
 #define BITS_RGBA_FIXED(r_, g_, b_, a_) \
 	.bits.r = r_, \
@@ -390,6 +394,14 @@ static const struct pixel_format_info pixel_format_table[] = {
 		.hsub = 2,
 		.vsub = 2,
 	},
+	{
+		DRM_FORMAT(P010),
+		SAMPLER_TYPE(EGL_TEXTURE_Y_UV_WL),
+		.num_planes = 2,
+		.chroma_order = ORDER_UV,
+		.hsub = 2,
+		.vsub = 2,
+	},
 	{
 		DRM_FORMAT(YUV422),
 		SAMPLER_TYPE(EGL_TEXTURE_Y_U_V_WL),
-- 
2.21.3

